src/package.js

```diff
@@ -0,0 +1,1106 @@
+        try {
+          const itemsBySelector = map['context-menu']
+          this.activationDisposables.add(this.contextMenuManager.add(itemsBySelector, validateSelectors))
+        } catch (error) {
+          if (error.code === 'EBADSELECTOR') {
+            error.message += ` in ${menuPath}`
+            error.stack += `\n  at ${menuPath}:1:1`
+          }
+          throw error
+        }
+      }
+    }
+
+    for (const [, map] of this.menus) {
+      if (map.menu) this.activationDisposables.add(this.menuManager.add(map.menu))
+    }
+
+    this.menusActivated = true
+  }
+
+  activateServices () {
+    let methodName, version, versions
+    for (var name in this.metadata.providedServices) {
+      ({versions} = this.metadata.providedServices[name])
+      const servicesByVersion = {}
+      for (version in versions) {
+        methodName = versions[version]
+        if (typeof this.mainModule[methodName] === 'function') {
+          servicesByVersion[version] = this.mainModule[methodName]()
+        }
+      }
+      this.activationDisposables.add(this.packageManager.serviceHub.provide(name, servicesByVersion))
+    }
+
+    for (name in this.metadata.consumedServices) {
+      ({versions} = this.metadata.consumedServices[name])
+      for (version in versions) {
+        methodName = versions[version]
+        if (typeof this.mainModule[methodName] === 'function') {
+          this.activationDisposables.add(this.packageManager.serviceHub.consume(name, version, this.mainModule[methodName].bind(this.mainModule)))
+        }
+      }
+    }
+  }
+
+  registerURIHandler () {
+    const handlerConfig = this.getURIHandler()
+    const methodName = handlerConfig && handlerConfig.method
+    if (methodName) {
+      this.uriHandlerSubscription = this.packageManager.registerURIHandlerForPackage(this.name, (...args) =>
+        this.handleURI(methodName, args)
+      )
+    }
+  }
+
+  unregisterURIHandler () {
+    if (this.uriHandlerSubscription) this.uriHandlerSubscription.dispose()
+  }
+
+  handleURI (methodName, args) {
+    this.activate().then(() => {
+      if (this.mainModule[methodName]) this.mainModule[methodName].apply(this.mainModule, args)
+    })
+    if (!this.mainActivated) this.activateNow()
+  }
+
+  registerTranspilerConfig () {
+    if (this.metadata.atomTranspilers) {
+      CompileCache.addTranspilerConfigForPath(this.path, this.name, this.metadata, this.metadata.atomTranspilers)
+    }
+  }
+
+  unregisterTranspilerConfig () {
+    if (this.metadata.atomTranspilers) {
+      CompileCache.removeTranspilerConfigForPath(this.path)
+    }
+  }
+
+  loadKeymaps () {
+    if (this.bundledPackage && this.packageManager.packagesCache[this.name]) {
+      this.keymaps = []
+      for (const keymapPath in this.packageManager.packagesCache[this.name].keymaps) {
+        const keymapObject = this.packageManager.packagesCache[this.name].keymaps[keymapPath]
+        this.keymaps.push([`core:${keymapPath}`, keymapObject])
```

Comment:
empty array + unconditional push seems like `.map` to me

Reply:
This is a loop over an object, so `for in` is convenient. An alternative would be `Object.keys(...).map`, though that would allocate an intermediate array of keys. Readability-wise, they seem pretty much equivalent.